pinot-common/src/main/java/com/linkedin/pinot/common/utils/time/DateTimeFieldSpecUtils.java

```diff
@@ -0,0 +1,393 @@
+    String[] formatTokens = format.split(COLON_SEPARATOR);
+    int size = Integer.valueOf(formatTokens[FORMAT_SIZE_POSITION]);
+    return size;
+  }
+
+  /**
+   * Extracts the column unit from the format of a dateTimeSpec
+   * (2nd token in colon separated format)
+   *
+   * eg: if format=5:MINUTES:EPOCH, will return MINUTES
+   * @return
+   */
+  @JsonIgnore
+  public static TimeUnit getColumnUnitFromFormat(String format) {
+    Preconditions.checkArgument(validFormat(format));
+    String[] formatTokens = format.split(COLON_SEPARATOR);
+    TimeUnit unit = TimeUnit.valueOf(formatTokens[FORMAT_UNIT_POSITION]);
+    return unit;
+  }
+
+  /**
+   * Extracts the TimeFormat from the format of a dateTimeSpec
+   * (3rd token in colon separated format)
+   *
+   * eg: if format=1:DAYS:EPOCH, will return EPOCH
+   * @return
+   */
+  @JsonIgnore
+  public static TimeFormat getTimeFormatFromFormat(String format) {
+    Preconditions.checkArgument(validFormat(format));
+    String[] formatTokens = format.split(COLON_SEPARATOR);
+    TimeFormat timeFormat = TimeFormat.valueOf(formatTokens[FORMAT_TIMEFORMAT_POSITION]);
+    return timeFormat;
+  }
+
+  /**
+   * Extracts the simmple date format pattern from the format of a dateTimeSpec
+   * (4th token in colon separated format in case of TimeFormat=SIMPLE_DATE_FORMAT)
+   *
+   * eg: if format=1:HOURS:EPOCH, will throw exception
+   * if format=1:HOURS:SIMPLE_DATE_FORMAT:yyyyMMddHH will return yyyyMMddHH
+   * @return
+   */
+  @JsonIgnore
+  public static String getSDFPatternFromFormat(String format) {
+    Preconditions.checkArgument(validFormat(format));
+    String pattern = null;
+    String[] formatTokens = format.split(COLON_SEPARATOR);
+    Preconditions.checkArgument(formatTokens.length == MAX_FORMAT_TOKENS, "SDF pattern does not exist for given format");
+    Preconditions.checkArgument(formatTokens[FORMAT_TIMEFORMAT_POSITION].equals(TimeFormat.SIMPLE_DATE_FORMAT.toString()),
+        "third token must be SIMPLE_DATE_FORMAT for an sdf pattern");
+    pattern = formatTokens[FORMAT_PATTERN_POSITION];
+    return pattern;
+  }
+
+  /**
+   * Given a timestamp in millis, convert it to the given format
+   * eg:
+   * 1) given dateTimeColumnValueMS = 1498892400000 and format=1:HOURS:EPOCH,
+   * dateTimeSpec.fromMillis(1498892400000) = 416359 (i.e. dateTimeColumnValueMS/(1000*60*60))
+   *
+   * 2) given dateTimeColumnValueMS = 1498892400000 and format=5:MINUTES:EPOCH,
+   * dateTimeSpec.fromMillis(1498892400000) = 4996308 (i.e. timeColumnValueMS/(1000*60*5))
+   *
+   * 3) given dateTimeColumnValueMS = 1498892400000 and format=1:DAYS:SIMPLE_DATE_FORMAT:yyyyMMdd,
+   * dateTimeSpec.fromMillis(1498892400000) = 20170701
+   * @param dateTimeColumnValueMS
+   * @param toFormat - the format in which to convert the millis value
+   * @return dateTime column value in dateTimeFieldSpec
+   */
+  @JsonIgnore
+  public static Object fromMillisToFormat(Long dateTimeColumnValueMS, String toFormat) {
```

Comment:
Why is the return type Object, and not a concrete type?

Reply:
The format can be simple date format, which is a string, or it can be epoch which is a long

Reply:
There's a few options:
1. Have different methods for different return types.
2. Create an interface.
3. If the caller knows the return type, you can have this return a generic type as opposed to Object.

Reply:
Fixed, with option 3